aws/resource_aws_cloudfront_function.go

```diff
@@ -0,0 +1,303 @@
+		FunctionCode: []byte(d.Get("code").(string)),
+		FunctionConfig: &cloudfront.FunctionConfig{
+			Comment: aws.String(d.Get("comment").(string)),
+			Runtime: aws.String(d.Get("runtime").(string)),
+		},
+		Name: aws.String(functionName),
+	}
+
+	CreateFunctionOutput, err := conn.CreateFunction(params)
+
+	if err != nil {
+		return fmt.Errorf("error creating Cloudfront Function: %w", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	d.Set("version", CreateFunctionOutput.ETag)
+	d.Set("arn", CreateFunctionOutput.FunctionSummary.FunctionMetadata.FunctionARN)
+	d.Set("last_modified", CreateFunctionOutput.FunctionSummary.FunctionMetadata.LastModifiedTime.Format(time.RFC3339))
+	d.Set("stage", CreateFunctionOutput.FunctionSummary.FunctionMetadata.Stage)
+	d.Set("status", CreateFunctionOutput.FunctionSummary.Status)
+
+	publish := d.Get("publish").(bool)
+	if publish {
+
+		params := &cloudfront.PublishFunctionInput{
+			Name:    aws.String(d.Get("name").(string)),
+			IfMatch: aws.String(d.Get("version").(string)),
+		}
+
+		log.Printf("[DEBUG] Publishing Cloudfront Function: %s", d.Id())
+
+		PublishFunctionOutput, err := conn.PublishFunction(params)
+		if err != nil {
+			return err
+		}
+		d.Set("status", PublishFunctionOutput.FunctionSummary.Status)
+		d.Set("last_modified", PublishFunctionOutput.FunctionSummary.FunctionMetadata.LastModifiedTime.Format(time.RFC3339))
+	}
+
+	return nil
+}
+
+// resourceAwsCloudFrontFunctionRead maps to:
+// GetFunction in the API / SDK
+func resourceAwsCloudFrontFunctionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	params := &cloudfront.GetFunctionInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Get Cloudfront Function: %s", d.Id())
+
+	GetFunctionOutput, err := conn.GetFunction(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudfront.ErrCodeNoSuchFunctionExists && !d.IsNewResource() {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("code", string(GetFunctionOutput.FunctionCode))
+
+	describeParams := &cloudfront.DescribeFunctionInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Fetching Cloudfront Function: %s", d.Id())
+
+	DescribeFunctionOutput, err := conn.DescribeFunction(describeParams)
+	if err != nil {
+		return err
+	}
+
+	d.Set("version", DescribeFunctionOutput.ETag)
+	d.Set("arn", DescribeFunctionOutput.FunctionSummary.FunctionMetadata.FunctionARN)
+	d.Set("last_modified", DescribeFunctionOutput.FunctionSummary.FunctionMetadata.LastModifiedTime.Format(time.RFC3339)) // 2006-01-02T15:04:05Z0700
+	d.Set("stage", DescribeFunctionOutput.FunctionSummary.FunctionMetadata.Stage)
+	d.Set("comment", DescribeFunctionOutput.FunctionSummary.FunctionConfig.Comment)
+	d.Set("runtime", DescribeFunctionOutput.FunctionSummary.FunctionConfig.Runtime)
+	d.Set("status", DescribeFunctionOutput.FunctionSummary.Status)
+	d.Set("publish", true)
```

Comment:
Don't set `publish`:

```suggestion
```

Reply:
Won't cause deleting it a constant drift?

Reply:
Should be fine because you have a `Default` value.

Reply:
OK. I've removed it yesterday